src/mixins/object_geometry.mixin.ts

```diff
@@ -1,870 +1,898 @@
-        return new Point(this.width, this.height).scalarAdd(this.strokeWidth);
-      },
+  /**
+   * calculate transform matrix that represents the current transformations from the
+   * object's properties, this matrix does not include the group transformation
+   * @return {TMat2D} transform matrix for the object
+   */
+  calcOwnMatrix(): TMat2D {
+    const key = this.transformMatrixKey(true),
+      cache = this.ownMatrixCache;
+    if (cache && cache.key === key) {
+      return cache.value;
+    }
+    const center = this.getRelativeCenterPoint(),
+      options = {
+        angle: this.angle,
+        translateX: center.x,
+        translateY: center.y,
+        scaleX: this.scaleX,
+        scaleY: this.scaleY,
+        skewX: this.skewX,
+        skewY: this.skewY,
+        flipX: this.flipX,
+        flipY: this.flipY,
+      },
+      value = composeMatrix(options);
+    this.ownMatrixCache = {
+      key,
+      value,
+    };
+    return value;
+  }
 
-      /**
-       * Calculate object bounding box dimensions from its properties scale, skew.
-       * @param {Object} [options]
-       * @param {Number} [options.scaleX]
-       * @param {Number} [options.scaleY]
-       * @param {Number} [options.skewX]
-       * @param {Number} [options.skewY]
-       * @private
-       * @returns {Point} dimensions
-       */
-      _getTransformedDimensions: function (options) {
-        options = Object.assign(
-          {
-            scaleX: this.scaleX,
-            scaleY: this.scaleY,
-            skewX: this.skewX,
-            skewY: this.skewY,
-            width: this.width,
-            height: this.height,
-            strokeWidth: this.strokeWidth,
-          },
-          options || {}
-        );
-        //  stroke is applied before/after transformations are applied according to `strokeUniform`
-        var preScalingStrokeValue,
-          postScalingStrokeValue,
-          strokeWidth = options.strokeWidth;
-        if (this.strokeUniform) {
-          preScalingStrokeValue = 0;
-          postScalingStrokeValue = strokeWidth;
-        } else {
-          preScalingStrokeValue = strokeWidth;
-          postScalingStrokeValue = 0;
-        }
-        var dimX = options.width + preScalingStrokeValue,
-          dimY = options.height + preScalingStrokeValue,
-          finalDimensions,
-          noSkew = options.skewX === 0 && options.skewY === 0;
-        if (noSkew) {
-          finalDimensions = new Point(
-            dimX * options.scaleX,
-            dimY * options.scaleY
-          );
-        } else {
-          var bbox = util.sizeAfterTransform(dimX, dimY, options);
-          finalDimensions = new Point(bbox.x, bbox.y);
-        }
-
-        return finalDimensions.scalarAdd(postScalingStrokeValue);
-      },
+  /**
+   * Calculate object dimensions from its properties
+   * @private
+   * @returns {Point} dimensions
+   */
+  _getNonTransformedDimensions(): Point {
```

Comment:
We should deprecate this method as well.
It is used only by `Object._renderBackground`

Reply:
And is weird in some context

Reply:
yes getNonTransformedDimensions is not used anymore as before. is no more the base for getTransformed.